src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+                case SpecialType.System_String:
+                    return StringConstantExpectedParameter.TryCreate(parameterSymbol, attributeData, out parameter);
+                default:
+                    parameter = null;
+                    return false;
+            }
+        }
+
+        private bool ValidateConstantExpectedParameter(IParameterSymbol parameterSymbol, out ImmutableArray<Diagnostic> diagnostics)
+        {
+            if (!TryGetConstantExpectedAttributeData(parameterSymbol, out var attributeData))
+            {
+                diagnostics = ImmutableArray<Diagnostic>.Empty;
+                return false;
+            }
+
+            switch (parameterSymbol.Type.SpecialType)
+            {
+                case SpecialType.System_Char:
+                    return UnmanagedHelper<char>.Validate(parameterSymbol, attributeData, char.MinValue, char.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Byte:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, byte.MinValue, byte.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UInt16:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, ushort.MinValue, ushort.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UInt32:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UInt64:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, ulong.MinValue, ulong.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UIntPtr:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_SByte:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, sbyte.MinValue, sbyte.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Int16:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, short.MinValue, short.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Int32:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Int64:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, long.MinValue, long.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_IntPtr:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, Helper, out diagnostics);
```

Comment:
@tannergooding In your opinion, is it a good practice to check for int.MinValue and int.MaxValue, since IntPtr has a platform-dependent size?